nautobot/docs/models/extras/dynamicgroup.md

```diff
@@ -1,7 +1,166 @@
+
+!!! note
+    The content type of a Dynamic Group cannot be modified once created, so take care in selecting this initially. This helps to reduce the possibility of inconsistent data and enforces the importance of thinking about the network data model when defining a new Dynamic Group.
+
+### Working with Dynamic Groups
+
+Dynamic Groups can be accessed from the primary Dynamic Groups landing page in the web interface under the _Organization > Dynamic Groups_ menu. From there you may view the list of available groups, search or filter the list, view or edit an individual group, or bulk delete groups. Additionally if a group's filter has matching members, the number of members may be clicked to take you to a filtered list view of those objects.
+
+Dynamic Groups cannot be imported nor can they be updated in bulk, as these operations would be complex and do not make sense in most cases.
+
+From an individual object's detail page, if it is a member of any groups, a "Dynamic Groups" tab will display in the navigation tabs. Clicking that tab will display all Dynamic Groups of which this object is a member.
+
+## Dynamic Groups and the REST API
+
+Dynamic Groups are fully supported by the API.
+
+### Specifying Filter Conditions
+
+Dynamic Groups are fairly straightforward however it is important to note that the `filter` field is a JSON field and it must be able to be used as valid query parameters for filtering objects of the corresponding content type.
+
+Consider, for example, the following Dynamic Group:
+
+```json
+{
+    "id": "9664758b-9de1-4b2b-87c2-8be40aa2238d",
+    "url": "http://localhost:6789/api/extras/dynamic-groups/9664758b-9de1-4b2b-87c2-8be40aa2238d/",
+    "name": "devices-a-star",
+    "slug": "devices-a-star",
+    "description": "Devices in sites starting with 'A'",
+    "content_type": "dcim.device",
+    "filter": {
+        "site": [
+            "ams01",
+            "ang01",
+            "atl01",
+            "atl02",
+            "azd01"
+        ]
+    },
+    "display": "devices-a-star"
+}
+```
+
+It is an error to provide any value other than a JSON object (`{}` or a Python dictionary) for the `filter` field. Additionally, most fields within the `filter` accept multiple values and must be represented as a JSON array (Python list). Certain fields take Boolean values (JSON `true`/`false`) or single numeric integers or character strings.
+
+For example, consider this filter:
+
+```json
+{
+  "has_interfaces": true,
+  "name": "ams01-edge-01"
+}
+```
+
+This would result in a Dynamic Group of a single Device with name "ams01-edge-01" if-and-only-if that device also has interfaces. While this likely wouldn't be all that useful of a filter for a Dynamic Group in practice, it illustrates that some fields take only one value instead of an array of values. It also underscores how important it is to think deliberately about defining your filter criteria.
```

Comment:
Somewhere in this document we need to explain the logic of multiple filter criteria within a single Dynamic Group, i.e. the idea that multiple filter values for a single field are *generally* ORed together (except for Tags!) while multiple distinct filter fields are generally ANDed together.

Reply:
Yeah that's in the latest updates. But your input is good to expand on it. This has turned out to be kind of not easy.